From f1137241f30689ffe38dd414c73f0b17bf08ecc8 Mon Sep 17 00:00:00 2001
From: popcornmix <popcornmix@gmail.com>
Date: Mon, 29 Apr 2019 19:16:16 +0100
Subject: [PATCH 439/703] Revert "staging: bcm2835-audio: use
 module_platform_driver() macro"

commit ed4c2e5dc4216d5dded502bfcf594d3984e6bccd upstream.

This reverts commit 786ced30fec053b27248ed5b24dcde61ed3f47f6.
---
 .../vc04_services/bcm2835-audio/bcm2835.c     | 20 ++++++++++++++++++-
 1 file changed, 19 insertions(+), 1 deletion(-)

--- a/drivers/staging/vc04_services/bcm2835-audio/bcm2835.c
+++ b/drivers/staging/vc04_services/bcm2835-audio/bcm2835.c
@@ -470,7 +470,25 @@ static struct platform_driver bcm2835_al
 		.of_match_table = snd_bcm2835_of_match_table,
 	},
 };
-module_platform_driver(bcm2835_alsa0_driver);
+
+static int bcm2835_alsa_device_init(void)
+{
+	int retval;
+
+	retval = platform_driver_register(&bcm2835_alsa0_driver);
+	if (retval)
+		pr_err("Error registering bcm2835_audio driver %d .\n", retval);
+
+	return retval;
+}
+
+static void bcm2835_alsa_device_exit(void)
+{
+	platform_driver_unregister(&bcm2835_alsa0_driver);
+}
+
+late_initcall(bcm2835_alsa_device_init);
+module_exit(bcm2835_alsa_device_exit);
 
 MODULE_AUTHOR("Dom Cobley");
 MODULE_DESCRIPTION("Alsa driver for BCM2835 chip");
